core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -90,6 +91,10 @@ private[spark] class EventLoggingListener(
    * Creates the log file in the configured log directory.
    */
   def start() {
+    val statusList = Option(fileSystem.listStatus(new Path(logBaseDir))).map(_.toSeq)
+      .getOrElse(Seq[FileStatus]())
+    EventLoggingListener.cleanRedundantLogFiles(sparkConf, fileSystem, statusList)
```

Comment:
No, don't touch this file. First because applying history server settings in Spark apps is weird.

Second because users don't even have permission to do this in a properly deployed Spark. Users can only delete their own log files.

Spark apps should only be concerned with their own state, not figuring out how to manage things left over by other Spark apps.